opnode/test/system_test.go

```diff
@@ -343,7 +343,66 @@ func TestSystemE2E(t *testing.T) {
 	seqBlock, err := l2Seq.BlockByNumber(context.Background(), receipt.BlockNumber)
 	require.Nil(t, err)
 	require.Equal(t, verifBlock.Hash(), seqBlock.Hash(), "Verifier and sequencer blocks not the same after including a batch tx")
+}
+
+func TestMintOnRevertedDeposit(t *testing.T) {
+	log.Root().SetHandler(log.DiscardHandler()) // Comment this out to see geth l1/l2 logs
+
+	cfg := defaultSystemConfig(t)
+
+	sys, err := cfg.start()
+	require.Nil(t, err, "Error starting up system")
+	defer sys.Close()
+
+	l1Client := sys.Clients["l1"]
+	l2Verif := sys.Clients["verifier"]
+
+	// Find deposit contract
+	depositContract, err := deposit.NewDeposit(cfg.DepositContractAddress, l1Client)
+	require.Nil(t, err)
+
+	// Initialize our sender address
+	ethPrivKey, err := sys.wallet.PrivateKey(accounts.Account{
+		URL: accounts.URL{
+			Path: transactorHDPath,
+		},
+	})
+	require.Nil(t, err)
+	opts, err := bind.NewKeyedTransactorWithChainID(ethPrivKey, cfg.L1ChainID)
+	fromAddr := crypto.PubkeyToAddress(ethPrivKey.PublicKey)
+	require.Nil(t, err)
+
+	toAddr := common.Address{0xff, 0xff}
```

Comment:
One nice check would to be sure that the `toAddr` balance is still 0 after the deposit (should mint if it has value, but should fail to transfer)